### DISCUSSION TITLE
Help a fåål

### DISCUSSION THREAD

**Discussion by NalleGang** [nickcoutsos/keymap-editor#203] [General]
Howdy, I just received my first keyboard like this and while I'm extremely excited, I'm also a big fool when it comes to tech but I didn't think it would be this hard to get ÄÅÖ on the keyboard when I bought it. Do you think there could be a way to somehow make those letters and other such available in the editor?

**Comment by nickcoutsos** [comment]
Hi @IrwinBadman 

I think the best way is going to be to include the appropriate headers from [zmk-locale-generator](https://github.com/joelspadin/zmk-locale-generator). I can't tell specifically which language you're after, so just taking German as an example, you could:

1. Go to the [latest release](https://github.com/joelspadin/zmk-locale-generator/releases/tag/v1.3.2) of zmk-locale-generator
2. Find your desired language, such as `keys_de.h`
3. Add the file to your repository under the `config/` folder
4. Include the file in your keymap by adding the line `#include "keys_de.h"` (somewhere near the top, with the other `#include ...` lines)

From this point you can use keycodes like `DE_A_UMLAUT` in your keymap, e.g. `&kp DE_A_UMLAUT`. When you load a keymap that contains the include statement in step 4 Keymap Editor will recognize the zmk-locale-generator header and the additional keycodes and symbols will be available for selection in the UI.

Note: behind the scenes these localized keycodes map to standard QWERTY keycodes, and the expectation is that in your Operating System you'll configure the input to match the intended language.

**Comment by arvidbjorkstrom** [comment]
That's nice to be able  to specifically add  special keys. But an even nicer thing for us with international keyboard layouts would be an option to have the labels on the keys be what we get when using them.

I have a  laptop  with a Nordic keyboard layout, and I have  my os set to a swedish keyboard layout. When I press the key on my Corne keyboard labelled `;` in the keyboard editor I get `Ö`, and when I press `/` I get `-` and so on for a lot of other keys.

As of now its really tricky to design your layout if you have a keyboard layout in you OS set to something other than US.

**Comment by NalleGang** [comment]
Whoops forgot to reply to this or close this but yeah figured out that "it just works" with :; etc. Thanks you. 

My keyboard journey has been wonderful
